diff --git a/bazel/dependency_imports.bzl b/bazel/dependency_imports.bzl
index 67fb6f7..fab7e7e 100644
--- a/bazel/dependency_imports.bzl
+++ b/bazel/dependency_imports.bzl
@@ -116,6 +116,11 @@ def envoy_dependency_imports(
         sum = "h1:Mb7Mrk043xzHgnRM88suvJFwzVrRfHEHJEl5/71CKw0=",
         version = "v0.34.0",
         build_external = "external",
+        build_directives = [
+            "gazelle:resolve go golang.org/x/text/secure/bidirule @org_golang_x_text//secure/bidirule",
+            "gazelle:resolve go golang.org/x/text/unicode/bidi @org_golang_x_text//unicode/bidi",
+            "gazelle:resolve go golang.org/x/text/unicode/norm @org_golang_x_text//unicode/norm",
+        ],
     )
     go_repository(
         name = "org_golang_x_text",
@@ -172,6 +177,11 @@ def envoy_dependency_imports(
         sum = "h1:EaGW2JJh15aKOejeuJ+wpFSHnbd7GE6Wvp3TsNhb6LY=",
         version = "v1.10.0",
         build_external = "external",
+        build_directives = [
+            "gazelle:resolve go golang.org/x/text/runes @org_golang_x_text//runes",
+            "gazelle:resolve go golang.org/x/text/transform @org_golang_x_text//transform",
+            "gazelle:resolve go golang.org/x/text/unicode/norm @org_golang_x_text//unicode/norm",
+        ],
     )
     go_repository(
         name = "com_github_lyft_protoc_gen_star_v2",
@@ -179,6 +189,10 @@ def envoy_dependency_imports(
         sum = "h1:sIXJOMrYnQZJu7OB7ANSF4MYri2fTEGIsRLz6LwI4xE=",
         version = "v2.0.4-0.20230330145011-496ad1ac90a4",
         build_external = "external",
+        build_directives = [
+            "gazelle:resolve go github.com/spf13/afero @com_github_spf13_afero//:afero",
+            "gazelle:resolve go golang.org/x/tools/imports @org_golang_x_tools//imports",
+        ],
     )
     go_repository(
         name = "com_github_iancoleman_strcase",
@@ -201,6 +215,9 @@ def envoy_dependency_imports(
         sum = "h1:TvGH1wof4H33rezVKWSpqKz5NXWg5VPuZ0uONDT6eb4=",
         version = "v1.3.0",
         build_external = "external",
+        build_directives = [
+            "gazelle:resolve go github.com/spf13/afero @com_github_spf13_afero//:afero",
+        ],
     )

     rules_proto_grpc_toolchains()
